### PR TITLE
47341_Trainee_finish_a_subject

### PR DESCRIPTION
**Pull request by Omid07**
screenshot from 2015-12-15 10 50 54

### CODE REVIEW

**Comment by asifbuetcse** [comment]
Ok

File: resources/views/trainee/subjects/index.blade.php [hunk 1]
```diff
@@ -8,6 +8,9 @@
             {!! $task->description !!}<br>
         @endforeach
         <br>
+        @if ($subject->pivot->is_finished == 0)
+        {!! Html::linkAction('Trainee\SubjectsController@edit', 'Finish Subject', [$subject->id]) !!}
```
Comment (dinhhoanglong91) [inline resources/views/trainee/subjects/index.blade.php]: When you change something, please sedn `POST` request. Action should be `Trainee\SubjectController@update` with method `POST`

File: app/Http/Controllers/Trainee/SubjectsController.php [hunk 2]
```diff
@@ -73,7 +75,10 @@ public function edit($id)
      */
     public function update(Request $request, $id)
     {
-        //
+        $course = \Auth::user()->courses()->where('course_user.is_currently_enrolled', 1)->first();
+        CourseSubject::whereSubjectId($id)->whereCourseId($course->id)->update(['is_finished' => 1, 'end_date' => date('Y-m-d h:i:sa')]);
```
Comment (dinhhoanglong91) [inline app/Http/Controllers/Trainee/SubjectsController.php]: `CourseSubject` doesn't belong to user. When user finishes a subject, it doesn't mean the all users finish this subject of this course. We can finish course's subject when all users finish this subject.
By the way, I don't know when to use this field `is_finished`

File: app/Http/Controllers/Trainee/SubjectsController.php [hunk 3]
```diff
@@ -73,7 +75,9 @@ public function edit($id)
      */
     public function update(Request $request, $id)
     {
-        //
+        $course = \Auth::user()->courses()->where('course_user.is_currently_enrolled', 1)->first();
+        SubjectUser::whereSubjectId($id)->whereCourseId($course->id)->update(['status' => 1]);
```
Comment (dinhhoanglong91) [inline app/Http/Controllers/Trainee/SubjectsController.php]: No user_id here, are you sure that you are editing right record?
Remember that multiple user can join multiple subjects of a course.

File: app/Http/Controllers/Trainee/SubjectsController.php [hunk 4]
```diff
@@ -16,8 +18,8 @@ class SubjectsController extends Controller
      */
     public function index()
     {
+        $subjects = \Auth::user()->subjects()->get();
```
Comment (dinhhoanglong91) [inline app/Http/Controllers/Trainee/SubjectsController.php]: We should work with subjects belongs to current course. So, you shouldn't change this `index()` function.

**Comment by dinhhoanglong91** [comment]
ok

**Pull request merged**